chainer/functions/array/reshape.py

```diff
@@ -51,12 +51,36 @@ def reshape(x, shape):
     """Reshapes an input variable without copy.
 
     Args:
-        x (~chainer.Variable): Input variable.
-        shape (tuple of ints): Target shape.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Input variable.
+        shape (:class:`tuple` of :class:`int` s):
+            The **size** of shape (**size** means the number of elements) must
+            be equal to that of original shape. One shape dimension can be -1.
+            In this case, the value is inferred from the length of the array
+            and remaining dimensions.
 
     Returns:
-        ~chainer.Variable: Variable that holds a reshaped version of the input
-            variable.
+        ~chainer.Variable:
+            Variable that holds a reshaped version of the input variable.
+
+    .. seealso:: :func:`numpy.reshape`, :func:`cupy.reshape`
+
+    .. admonition:: Example
+
+        >>> x = np.array([[1, 2, 3, 4], [5, 6, 7, 8]])
+        >>> y = F.reshape(x, (8,))
+        >>> y.shape
+        (8,)
+        >>> y.data
+        array([1, 2, 3, 4, 5, 6, 7, 8])
+        >>> y = F.reshape(x, (4,-1))
+        >>> y.shape
+        (4, 2)
+        >>> y.data
+        array([[1, 2],
+               [3, 4],
+               [5, 6],
+               [7, 8]])
```

Comment:
you mean that the case like this?
```
>>> x = np.array([[1, 2, 3, 4], [5, 6, 7, 8]])
>>> y = F.reshape(x, (4, 3))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kumezawa/dev/chainer/chainer/functions/array/reshape.py", line 86, in reshape
    return Reshape(shape)(x)
  File "/home/kumezawa/dev/chainer/chainer/function.py", line 189, in __call__
    self._check_data_type_forward(in_data)
  File "/home/kumezawa/dev/chainer/chainer/function.py", line 273, in _check_data_type_forward
    self.check_type_forward(in_type)
  File "/home/kumezawa/dev/chainer/chainer/functions/array/reshape.py", line 32, in check_type_forward
    type_check.prod(x_type.shape) == type_check.prod(self.shape))
  File "/home/kumezawa/dev/chainer/chainer/utils/type_check.py", line 487, in expect
    expr.expect()
  File "/home/kumezawa/dev/chainer/chainer/utils/type_check.py", line 449, in expect
    '{0} {1} {2}'.format(left, self.inv, right))
chainer.utils.type_check.InvalidType: 
Invalid operation is performed in: Reshape (Forward)

Expect: prod(in_types[0].shape) == prod((4, 3))
Actual: 8 != 12
```

I can add the example, but the error message is a little verbose.